### PR TITLE
implement dictionary of variable names and getVar

### PR DESCRIPTION
**Pull request by mlubin**
Needs docs. Doesn't seem like there should be any performance penalty, but worth double checking.
@davidanthoff

### CODE REVIEW

**Comment by IainNZ** [comment]
Seems nice and simple!

File: src/JuMP.jl [hunk 1]
```diff
@@ -730,6 +733,28 @@ function Variable(m::Model,lower::Number,upper::Number,cat::Symbol,objcoef::Numb
...
+# handle dictionary of variables
+function registervar(m::Model, varname::Symbol, value)
+    if haskey(m.varDict, varname)
+        m.varDict[varname] = nothing # indicate duplicate variable
+    else
+        m.varDict[varname] = value
+    end
+    return value
+end
+registervar(m::Model, varname, value) = value # variable name isn't a simple symbol, ignore
```
Comment (joehuchette) [inline src/JuMP.jl]: When would this get called?

**Comment by joehuchette** [comment]
:+1:

File: test/variable.jl [hunk 2]
```diff
@@ -27,6 +27,15 @@ facts("[variable] constructors") do
...
     @fact getUpper(x[-4,"Green"]) => 5.5
     @fact getValue(x[-3,"Blue"]) => -2
+    @fact isequal(getVar(mcon, :lbonly),lbonly) => true
+    @fact isequal(getVar(mcon, :ubonly),ubonly) => true
+    @fact isequal(getVar(mcon, :onerangeub)[-7],onerangeub[-7]) => true
+    @defVar(mcon, lbonly)
+    @fact_throws ErrorException getVar(mcon, :lbonly)
+    @fact_throws ErrorException getVar(mcon, :foo)
+    d = Dict()
+    @defVar(mcon, d["bar"][1:10] == 1)
```
Comment (mlubin) [inline test/variable.jl]: @joehuchette, it would get called here

Comment (joehuchette) [inline test/variable.jl]: Hmm. Maybe instead of symbols you could use strings, in which case this would work? Not sure it's worth it as this is kind of a corner case.

Comment (mlubin) [inline test/variable.jl]: I don't think there's much of a need to support that in the first pass

Comment (joehuchette) [inline test/variable.jl]: Agreed, we could always add it in without breaking anything later if there's a need.

**Comment by davidanthoff** [comment]
Looks perfect, this will help us a lot :)

**Comment by mlubin** [comment]
Updated ``NEWS`` and added a line in the docs.

**Comment by tkelman** [comment]
This looks pretty handy indeed.
